### PR TITLE
Feature/ia 5/chat exercise

### PR DESCRIPTION
**Pull request by MiguelAskar**
This gif  shows the simple chat working:

### CODE REVIEW

File: SimpleChatChallenge/Manager/MessageManager.swift [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+//  SimpleChatChallenge
+//
+//  Created by Miguel Angel Askar Rodriguez on 29/07/21.
+//
+
+import Foundation
+
+class MessageManager {
+    
+    static let shared = MessageManager()
```
Comment (pablocalcagnino) [inline SimpleChatChallenge/Manager/MessageManager.swift:12]: avoid using singletons

File: SimpleChatChallenge/Utils/Serialization.swift [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+//  Serialization.swift
+//  SimpleChatChallenge
+//
+//  Created by Miguel Angel Askar Rodriguez on 28/07/21.
+//
+
+import Foundation
+
+struct Serialization {
+    static func load<T: Codable>(fromJSON resource: String) -> T? {
```
Comment (pablocalcagnino) [inline SimpleChatChallenge/Utils/Serialization.swift:11]: good use of generics, have you read about them?

**Review by pablocalcagnino — APPROVED** [review]

**Review by cmorales-globant — APPROVED** [review]
Nice Job!

File: SimpleChatChallenge/Utils/Serialization.swift (hunk 2 above)
Comment (MiguelAskar) [reply SimpleChatChallenge/Utils/Serialization.swift:11]: Actually, the credit for this code goes to my mentor, Camilo. The code is his, but I made sure I understood.

**Pull request merged**